core/blockchain.go

```diff
@@ -1619,6 +1620,110 @@ func (bc *BlockChain) AddExternalBlock(block *types.ExternalBlock) error {
 	return nil
 }
 
+// ReOrgRollBack compares the difficulty of the newchain and oldchain. Rolls back
+// the current header to the position where the reorg took place in a higher context
+func (bc *BlockChain) ReOrgRollBack(header *types.Header) error {
+	var (
+		deletedTxs  types.Transactions
+		deletedLogs [][]*types.Log
+
+		// collectLogs collects the logs that were generated or removed during
+		// the processing of the block that corresponds with the given hash.
+		// These logs are later announced as deleted or reborn
+		collectLogs = func(hash common.Hash) {
```

Comment:
What is the purpose of collecting the logs for the passed in header?

Reply:
we are collecting logs for all the blocks that are about to get deleted and we send this log to the deleted logs feed which goes back to mempool